hazelcast/src/main/java/com/hazelcast/util/BufferingInputStream.java

```diff
@@ -0,0 +1,81 @@
+/*
```

Comment:
Yes, that was my plan. This is why I added the `BufferingInputStream` in that common package and just removed the fixed `BUFFER_SIZE` from it. If this get's merged, we should be able to re-use the class in HotRestart. I have to look at the other methods which I borrowed from the other HotRestart classes though.